python/dgl/heterograph.py

```diff
@@ -0,0 +1,1214 @@
+        6
+        """
+        pass
+
+    def node_attr_schemes(self, ntype):
+        """Return the node feature schemes for a given node type.
+
+        Each feature scheme is a named tuple that stores the shape and data type
+        of the node feature
+
+        Parameters
+        ----------
+        ntype : str
+            The node type
+
+        Returns
+        -------
+        dict of str to schemes
+            The schemes of node feature columns.
+        """
+        pass
+
+    def edge_attr_schemes(self, etype):
+        """Return the edge feature schemes for a given edge type.
+
+        Each feature scheme is a named tuple that stores the shape and data type
+        of the edge feature
+
+        Parameters
+        ----------
+        etype : tuple[str, str, str]
+            The edge type, characterized by a triplet of source type name,
+            destination type name, and edge type name.
+
+        Returns
+        -------
+        dict of str to schemes
+            The schemes of node feature columns.
+        """
+        pass
+
+    def set_n_initializer(self, ntype, initializer, field=None):
+        """Set the initializer for empty node features of given type.
+
+        Initializer is a callable that returns a tensor given the shape, data type
+        and device context.
+
+        When a subset of the nodes are assigned a new feature, initializer is
+        used to create feature for rest of the nodes.
+
+        Parameters
+        ----------
+        ntype : str
+            The node type name.
+        initializer : callable
+            The initializer.
+        field : str, optional
+            The feature field name. Default is set an initializer for all the
+            feature fields.
+        """
+        pass
+
+    def set_e_initializer(self, etype, initializer, field=None):
+        """Set the initializer for empty edge features of given type.
+
+        Initializer is a callable that returns a tensor given the shape, data
+        type and device context.
+
+        When a subset of the edges are assigned a new feature, initializer is
+        used to create feature for rest of the edges.
+
+        Parameters
+        ----------
+        etype : tuple[str, str, str]
+            The edge type, characterized by a triplet of source type name,
+            destination type name, and edge type name.
+        initializer : callable
+            The initializer.
+        field : str, optional
+            The feature field name. Default is set an initializer for all the
+            feature fields.
+        """
+        pass
+
+    @property
+    def nodes(self):
+        """Return a node view that can used to set/get feature data of a
+        single node type.
+
+        Notes
+        -----
+        An error is raised if the graph contains multiple node types.  Use
+
+            g[ntype]
+
+        to select nodes with type ``ntype``.
+        """
```

Comment:
can we add a node type argument here? if `g.nodes` doesn't work for heterogeneous graphs,  `g['user', 'item'].nodes` doesn't work either. the only way that works is `g['user'].nodes`?

Reply:
Only `g['user'].nodes` is allowed; `g['user', 'item'].nodes` would throw an error.

Reply:
Wait, `g['user', 'item'].nodes['user']` is allowed? No?

Reply:
No.  `nodes[]` takes in IDs.